.github/workflows/go.yaml

```diff
@@ -30,6 +34,18 @@ jobs:
         run: 'echo head_ref: ${{ github.head_ref }}, ref: ${{ github.ref }}, go sdk version: ${{ inputs.version }}'
         working-directory: '.'
 
+      - name: Download docker artifacts
+        if: ${{ inputs.docker-image-artifact-name }}
+        uses: actions/download-artifact@v3
```

Comment:
Are we expecting to build and upload a docker image for Temporal server on every PR? Ideally we could provide a server _commit/ref_ and build the server at test time. Or can we share the docker container built on the server side with this somehow without having to push then download? Or am I misunderstanding?

Granted I think the same about the constant SDK features containers too. We may be adding unnecessary steps containerizing some of these things.

Reply:
You're misunderstanding. The server triggered workflows build a docker image which is used internally as an artifact. Nothing is pushed.

Building the docker image takes essentially an equivalent amount of time to just building the server, but makes using the built artifacts dramatically easier across parallel runs.